api/api-roles-config-v3/src/main/java/com/thoughtworks/go/apiv3/rolesconfig/representers/DirectiveRepresenter.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2019 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv3.rolesconfig.representers;
+
+import com.google.gson.JsonParseException;
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.config.policy.Allow;
+import com.thoughtworks.go.config.policy.Deny;
+import com.thoughtworks.go.config.policy.Directive;
+
+public class DirectiveRepresenter {
+
+    public static void toJSON(OutputWriter jsonWriter, Directive directive) {
+        jsonWriter.add("permission", directive.getDirectiveType().type())
+                .add("action", directive.action())
+                .add("type", directive.type())
+                .add("resource", directive.resource());
+    }
+
+    public static Directive fromJSON(JsonReader jsonReader) {
+        String permission = jsonReader.getString("permission");
+        if (permission.equals("allow")) {
+            return new Allow(jsonReader.getString("action"), jsonReader.getString("type"), jsonReader.getString("resource"));
+        } else if (permission.equals("deny")) {
+            return new Deny(jsonReader.getString("action"), jsonReader.getString("type"), jsonReader.getStringOrDefault("resource", null));
+        } else {
+            throw new JsonParseException(String.format("Invalid permission '%s'. It has to be one of 'allow' or 'deny'.", permission));
+        }
```

Comment:
Few suggestions:
 - can we create individuals vars for action, type, etc and then create `Allow` inside `if`. 
    Reason: better readability 
- instead of throwing `JsonParseException`, can we create a dummy `UnknownDirective` and add a config error to the same. This will allow us to return the original payload with errors on them.

Reference: https://github.com/gocd/gocd/blob/master/api/api-secret-configs-v1/src/main/java/com/thoughtworks/go/apiv1/secretconfigs/representers/RulesRepresenter.java#L67-L88

Reply:
@kritika-singh3 -- Fixed the `JsonParseException` as suggested.
The API gives the following response in case of invalid permission:

```json
{
  "message" : "Save failed. An error occurred while saving the role config. Please check the logs for more information.",
  "data" : {
    "policy" : [ {
      "errors" : {
        "permission" : [ "Invalid permission, must be either 'allow' or 'deny'." ]
      },
      "permission" : "awllow",
	  ...
    } ],
	...
  }
}
```